### PR TITLE
Add metric support to Feedback Listener

### PR DESCRIPTION
**Pull request by mbotarro**
This PR:

Adds support to `StatsReporter` in the `Listener`, `Broker` and `InvalidTokenHandler`.

The `Listener` periodically flushes the pipeline channels’ size as a gauge metric. The flush interval can be set using the new env variable `PUSHER_STATS_FLUSH_S`.

The `InvalidTokenHandler` sends metrics related to the tokens deletion. Its unit tests have been updated using a `StatsD` mock.

Beyond that, the following modifications have been introduced: 

* A `Count` method has been added in the `Statsd` Interface and its mock has been updated since there was a `Count` field that caused a conflict
* Two new methods have been introduced to `StatsD` and `StatsReporter` interface to send generic `Gauge` and `Count` metrics with a game and platform as tags
* There was en error where the PG mock wasn’t able to return the correct `RowsAffected` in the result. To solve that,  the mock has been updated to put a `null` character at the end of the query in a way that the `NewResult` function is now able to correctly parse it.
* A `common.go` file was created to manage the reporters initialisation and use
* The Listener’s Stop method has been removed to prevent from closing the stopChannel twice

### CODE REVIEW

**Comment by cscatolini** [comment]
@mbotarro actually the env var is `PUSHER_STATS_FLUSH_S` since all env vars are prefixed with `PUSHER_`

File: feedback/invalid_token.go [hunk 1]
```diff
@@ -213,16 +221,38 @@ func (i *InvalidTokenHandler) deleteTokensFromGame(tokens []string, game, platfo
...
+
+		return nil
+	}
+
+	if res.RowsAffected() != len(tokens) {
+		statsReporterReportMetricCount(i.StatsReporter,
+			MetricsTokensDeleteNonexistent, int64(len(tokens)-res.RowsAffected()),
+			game, platform)
+
+		return nil
```
Comment (cscatolini) [inline feedback/invalid_token.go]: Here we should also report MetricsTokensDeleteSuccess with count `res.RowsAffected()` before returning, right?

File: feedback/listener.go [hunk 2]
```diff
@@ -80,13 +93,13 @@ func (l *Listener) configure() error {
 	}
 	l.Queue = q
 
-	broker, err := NewBroker(l.Logger, l.Config, q.MessagesChannel(), l.Queue.PendingMessagesWaitGroup())
+	broker, err := NewBroker(l.Logger, l.Config, nil, q.MessagesChannel(), l.Queue.PendingMessagesWaitGroup())
```
Comment (cscatolini) [inline feedback/listener.go]: I think we'd better send StatsReporters instead of nil in here.

File: feedback/listener.go [hunk 3]
```diff
@@ -80,13 +93,13 @@ func (l *Listener) configure() error {
...
 	if err != nil {
 		return fmt.Errorf("error creating new broker: %s", err.Error())
 	}
 	l.Broker = broker
 
-	handler, err := NewInvalidTokenHandler(l.Logger, l.Config, l.Broker.InvalidTokenOutChan)
+	handler, err := NewInvalidTokenHandler(l.Logger, l.Config, nil, l.Broker.InvalidTokenOutChan)
```
Comment (cscatolini) [inline feedback/listener.go]: Same as above

File: feedback/invalid_token.go (hunk 1 above)
Comment (mbotarro) [reply feedback/invalid_token.go]: Yep, you're right! I'll change that.

**Pull request merged**